From e97219847c56b2fb1f7c624d559465b8754fd8ae Mon Sep 17 00:00:00 2001
From: Matthias Schiffer <mschiffer@universe-factory.net>
Date: Mon, 10 Jul 2017 18:03:39 +0200
Subject: [PATCH 38/69] ipq806x: upgrade: fix RAMFS_COPY_*

RAMFS_COPY_* are moved to platform.sh toplevel. The unneeded
linksys_preupgrade function is removed, and the nand_do_upgrade call is
moved to platform_do_upgrade.

Fixes: 30f61a34b4cf "base-files: always use staged sysupgrade"
Signed-off-by: Matthias Schiffer <mschiffer@universe-factory.net>
---
 target/linux/ipq806x/base-files/lib/upgrade/linksys.sh  |  9 ---------
 target/linux/ipq806x/base-files/lib/upgrade/platform.sh | 17 +++++------------
 2 files changed, 5 insertions(+), 21 deletions(-)

diff --git a/target/linux/ipq806x/base-files/lib/upgrade/linksys.sh b/target/linux/ipq806x/base-files/lib/upgrade/linksys.sh
index 12f25efdae..0234ce0a51 100644
--- a/target/linux/ipq806x/base-files/lib/upgrade/linksys.sh
+++ b/target/linux/ipq806x/base-files/lib/upgrade/linksys.sh
@@ -99,12 +99,3 @@ platform_do_upgrade_linksys() {
 		fi
 	}
 }
-
-linksys_preupgrade() {
-	export RAMFS_COPY_BIN="${RAMFS_COPY_BIN} /usr/sbin/fw_printenv /usr/sbin/fw_setenv"
-	export RAMFS_COPY_BIN="${RAMFS_COPY_BIN} /bin/mkdir /bin/touch"
-	export RAMFS_COPY_DATA="${RAMFS_COPY_DATA} /etc/fw_env.config /var/lock/fw_printenv.lock"
-
-}
-
-append sysupgrade_pre_upgrade linksys_preupgrade
diff --git a/target/linux/ipq806x/base-files/lib/upgrade/platform.sh b/target/linux/ipq806x/base-files/lib/upgrade/platform.sh
index c3741f9ff2..6455b57cfb 100644
--- a/target/linux/ipq806x/base-files/lib/upgrade/platform.sh
+++ b/target/linux/ipq806x/base-files/lib/upgrade/platform.sh
@@ -3,11 +3,14 @@
 PART_NAME=firmware
 REQUIRE_IMAGE_METADATA=1
 
+RAMFS_COPY_BIN='/usr/sbin/fw_printenv /usr/sbin/fw_setenv'
+RAMFS_COPY_DATA='/etc/fw_env.config /var/lock/fw_printenv.lock'
+
 platform_check_image() {
 	return 0;
 }
 
-platform_pre_upgrade() {
+platform_do_upgrade() {
 	local board=$(ipq806x_board_name)
 
 	case "$board" in
@@ -17,18 +20,8 @@ platform_pre_upgrade() {
 	r7500 |\
 	r7500v2 |\
 	r7800)
-		nand_do_upgrade "$1"
-		;;
-	ea8500)
-		linksys_preupgrade "$1"
+		nand_do_upgrade "$ARGV"
 		;;
-	esac
-}
-
-platform_do_upgrade() {
-	local board=$(ipq806x_board_name)
-
-	case "$board" in
 	c2600)
 		PART_NAME="os-image:rootfs"
 		MTD_CONFIG_ARGS="-s 0x200000"
-- 
2.16.2

